test/integration/targets/git/tasks/submodules.yml

```diff
@@ -14,6 +14,90 @@
 # Repository III for a second submodule (repo_submodule2)
 #   Has 1 file checked in
 
+- name: SUBMODULES | setup
+  block:
+    - name: SUBMODULES | setup | start git daemon and create repositories
+      include_role: name=test_utils/roles/git-repos
+      vars:
+        repositories:
+          - test_submodules_newer.git
+          - test_submodules_subm1.git
+          - test_submodules_subm2.git
+        exec:
+          - 'echo initial > README.md'
+          - 'git add README.md && git commit -m "Initial commit"'
+          - 'git tag v0.1'
+          - 'git push origin {{ git_default_branch }} --tags'
+
+    - name: SUBMODULES | setup | add submodule1
+      include_role: name=test_utils/roles/git-repos
+      vars:
+        repositories:
+          - test_submodules_newer.git
+        exec:
+          - 'git submodule add --name submodule1 git://localhost/test_submodules_subm1.git submodule1'
```

Comment:
Usually, doing the following works just fine:
```diff
-          - 'git submodule add --name submodule1 git://localhost/test_submodules_subm1.git submodule1'
+          - 'git submodule add --name submodule1 file://path/to/bare/repo/test_submodules_subm1.git submodule1'
```
Or maybe
```diff
-          - 'git submodule add --name submodule1 git://localhost/test_submodules_subm1.git submodule1'
+          - 'git submodule add --name submodule1 ../relative/path/to/bare/repo/test_submodules_subm1.git submodule1'
```
*(I don't remember which one works better, but I did use this in the past)*

Reply:
The goal of this PR is to replace dependencies on external Github repos that aren't managed by Ansible, and not be limited to 1 protocol since some features cannot be tested with file://.

The current tests use absolute URLs, so that's why these are absolute. I could use file:// for these specifically, but a git server is necessary to fully test relative submodules (which can be relative to the repo or origin remote), and because I know that submodules relative to the remote are broken (see https://github.com/ansible/ansible/issues/83146), I added the git daemon in this PR to make sure it works with our existing test cases, before I add more complex submodule tests.

Reply:
Ah, I see. Thanks for explaining!